src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.CompilationTracker.cs

```diff
@@ -583,9 +583,7 @@ private async Task<Compilation> GetOrBuildDeclarationCompilationAsync(SolutionSe
                     }
 
                     compilation = compilation.AddSyntaxTrees(trees);
-                    trees.Free();
-
-                    WriteState(new FullDeclarationState(compilation, TextDocumentStates<SourceGeneratedDocumentState>.Empty, generatorDriver: null, generatedDocumentsAreFinal: false), solutionServices);
+                    WriteState(new AllSyntaxTreesParsedState(compilation, generatedDocuments, generatorDriver, generatedDocumentsAreFinal), solutionServices);
```

Comment:
Hmm, good catch to make sure we don't lose these -- I can't tell if we had a bug here before (where in some obscure cases we might end up losing prior generated documents), or whether we couldn't have gotten here before once we had generated everything.